### ISSUE TITLE
Low PCC for ttnn.multiply after ttnn.floor_div with program cache enabled

### ISSUE THREAD

**Issue by jmalone-tt** [tenstorrent/tt-metal#21086]
### Describe the bug

While enabling program cache for the Pytorch project, an issue with low PCC for a certain op sequence was found. In particular, the PCC is low for ttnn.multiply followed by ttnn.floor_div where both broadcast dimensions of an op. Repro steps are below.

### Steps to reproduce the issue

Run the following code, and you should see an assert about low pcc (amount varies due to random inputs):
```python
import torch
import ttnn

import numpy as np

def comp_pcc(golden, calculated, pcc=0.99):
    golden = torch.Tensor(golden)
    calculated = torch.Tensor(calculated)

    if golden.dtype != calculated.dtype:
        calculated = calculated.type(golden.dtype)

    if torch.all(torch.isnan(golden)) and torch.all(torch.isnan(calculated)):
        # logger.warning("Both tensors are 'nan'")
        return True, 1.0

    if torch.all(torch.isnan(golden)) or torch.all(torch.isnan(calculated)):
        # logger.error("One tensor is all nan, the other is not.")
        return False, 0.0

    # Test if either is completely zero
    if torch.any(golden.bool()) != torch.any(calculated.bool()):
        # logger.error("One tensor is all zero")
        return False, 0.0

    # For now, mask all infs and nans so that we check the rest... TODO
    golden = golden.clone()
    golden[
        torch.logical_or(
            torch.isnan(golden),
            torch.logical_or(torch.isinf(golden), torch.isneginf(golden)),
        )
    ] = 0
    calculated = calculated.clone()
    calculated[
        torch.logical_or(
            torch.isnan(calculated),
            torch.logical_or(torch.isinf(calculated), torch.isneginf(calculated)),
        )
    ] = 0

    if torch.equal(golden, calculated):
        return True, 1.0

    if golden.dtype == torch.bfloat16:
        golden = golden.type(torch.float32)
        calculated = calculated.type(torch.float32)
    cal_pcc = np.min(
        np.ma.corrcoef(
            np.ma.masked_invalid(torch.squeeze(golden).detach().numpy()).flatten(),
            np.ma.masked_invalid(torch.squeeze(calculated).detach().numpy()).flatten(),
        )
    )

    if isinstance(cal_pcc, np.ma.core.MaskedConstant):
        return True, 1.0

    return cal_pcc >= pcc, cal_pcc

def assert_with_pcc(expected_pytorch_result, actual_pytorch_result, pcc=0.999):
    if isinstance(expected_pytorch_result, ttnn.Tensor):
        expected_pytorch_result = ttnn.to_torch(expected_pytorch_result, dtype=torch.bfloat16)

    assert list(expected_pytorch_result.shape) == list(
        actual_pytorch_result.shape
    ), f"list(expected_pytorch_result.shape)={list(expected_pytorch_result.shape)} vs list(actual_pytorch_result.shape)={list(actual_pytorch_result.shape)}"
    pcc_passed, pcc_message = comp_pcc(expected_pytorch_result, actual_pytorch_result, pcc)
    assert pcc_passed, f"pcc check failed (expected {pcc} but got {pcc_message})"

    return pcc_passed, pcc_message

def floor_divide_forward(device, arg0_1, arg1_1):
    ttnn_from_torch = ttnn.from_torch(arg0_1, device = device, layout = ttnn.TILE_LAYOUT, dtype = ttnn.bfloat16)
    ttnn_from_torch_1 = ttnn.from_torch(arg1_1, device = device, layout = ttnn.TILE_LAYOUT, dtype = ttnn.bfloat16)
    ttnn_floor_div = ttnn.floor_div(ttnn_from_torch, ttnn_from_torch_1)
    ttnn_to_torch = ttnn.to_torch(ttnn_floor_div, dtype = torch.bfloat16)
    return ttnn_to_torch


def reduced_op_forward(device, arg0_1, arg1_1):
    ttnn_from_torch = ttnn.from_torch(arg0_1, device = device, layout = ttnn.TILE_LAYOUT, dtype = ttnn.bfloat16)
    ttnn_from_torch_1 = ttnn.from_torch(arg1_1, device = device, layout = ttnn.TILE_LAYOUT, dtype = ttnn.bfloat16)

    ttnn_reshape = ttnn.reshape(ttnn_from_torch_1, (1, 100, 1, 1))
    ttnn_multiply = ttnn.multiply(ttnn_from_torch, ttnn_reshape)

    reshaped_arg1_1 = torch.reshape(arg1_1, (1, 100, 1, 1))
    torch_multiply = torch.multiply(arg0_1, reshaped_arg1_1)

    assert_with_pcc(ttnn_multiply, torch_multiply, pcc=0.9)

if __name__ == "__main__":
    # open device
    device = ttnn.open_device(device_id=0, l1_small_size=16384)
    ttnn.enable_program_cache(device)

    # run floor_divide
    numerator = torch.randint(1, 15, (1, 12, 9, 9)).to(torch.bfloat16)
    denominator = torch.randint(1, 15, ()).to(torch.bfloat16)
    floor_divide_forward(device, numerator, denominator)

    # run batch_norm_inference
    input_1 = torch.randn((1, 100, 14, 14), dtype=torch.bfloat16)
    input_2 = torch.randn(100, dtype=torch.bfloat16)

    # run to test pcc
    reduced_op_forward(device, input_1, input_2)

    # close device
    ttnn.close_device(device)
```

### Expected behavior

The reproduction test should pass without asserting.

### Please complete the following environment information

Ubuntu 22.04
python 3.10
tt-metal RC60

**Comment by bbradelTT** [comment]
This is eltwise. Assigned to @cmaryanTT to re-assign.

**Comment by Aswinmcw** [comment]
Hi @cmaryanTT , while debugging this issue, below is our findings,

1. Using binary ng version of multiply will make the test pass without any issues
2. In legacy binary version, if we remove the hash computation part, the test will pass
3. pcc drop is happening if the HW bcast kernel is called so the issue is most likely to be in bcast kernel
4. regarding the test mentioned above it is not related to floor_div op, calling the multiply back to back causes this issue while other add sub div wont shows this failure

**Comment by Aswinmcw** [comment]
It uses scalar broadcasting and currently falling back to legacy version, need to force it to use binary_ng to make the test pass

**Comment by cmaryanTT** [comment]
@dchenTT - Another scalar broadcast case to fix up.

**Comment by dchenTT** [comment]
Fixed by https://github.com/tenstorrent/tt-metal/pull/23686 No flag is required.

**Comment by dchenTT** [comment]
The legacy generally speaking works for scalar bcast. However, the accuracy may be a concern. Since it uses FPU, not SFPU, for float32, the accuracy may be lower than expectl. For bfloat16, accuracy should be fine. I am not sure what the problem is if bfloat16 also have low accuracy. We do not want to spend much time on legacy if binary_ng just works well.

**Comment by fahmedchanTT** [comment]
@jmalone-tt from @dchenTT 's comment can we close or rework this case ?

**Comment by jmalone-tt** [comment]
We should be good to close the case, thanks for checking!
